skrub/_agg_joiner.py

```diff
@@ -0,0 +1,592 @@
+from copy import deepcopy
+
+import numpy as np
+import pandas as pd
+from sklearn.base import BaseEstimator, TransformerMixin
+from sklearn.utils.multiclass import type_of_target
+from sklearn.utils.validation import check_is_fitted
+
+import skrub._agg_pandas as agg_pd
+import skrub._agg_polars as agg_pl
+
+NUM_OPS = ["sum", "mean", "std", "min", "max", "hist", "value_counts"]
+CATEG_OPS = ["mode", "count", "value_counts"]
+ALL_OPS = NUM_OPS + CATEG_OPS
+
+
+def split_num_categ_ops(agg_ops):
+    """Separate aggregagor operators input
+    by their type.
+
+    Parameters
+    ----------
+    agg_ops : list of str,
+        The input operators names.
+
+    Returns
+    -------
+    num_ops, categ_ops : Tuple of List of str
+        List of operator names
+    """
+    num_ops, categ_ops = [], []
+    for op_name in agg_ops:
+        # hist(5) -> hist
+        op_root = op_name.split("(")[0]
+        if op_root in NUM_OPS:
+            num_ops.append(op_name)
+        if op_root in CATEG_OPS:
+            categ_ops.append(op_name)
+        if op_root not in ALL_OPS:
+            raise ValueError(f"'agg_ops' options are {ALL_OPS}, got: '{op_name}'.")
+    return num_ops, categ_ops
+
+
+def get_namespace(dataframes):
```

Comment:
Thanks for the comments on typing. I prefer doing these last because they are subject to change when the API changes. So it's like painting the house after it's fully built ;)